Code.gs

```diff
@@ -1,69 +1,206 @@
-  
-  var CompanyCol = 1
-  var TimesheetCol = 17
-  
-  //IDs
-  var orgId = '1IwctVagVOgmlmGbJt0atVQPkyTp_ZKocoRkuQa4cEtU'
-  //var timesheetId = '1wk717ZUEo50crwZr8fUXdutD1YproOzi4SughyjHwSU'
-  var dociId = '12lKDk6A-P2-JGo4vhz1fE_33vC_MvxC1hbcUpfV5R2o' 
- 
-  //Open the Journal
-  var journal = DocumentApp.openById(dociId)
-  var body = journal.getBody()
-  
-  if (journal == '') {
-    Logger.log('Invalid Journal ID')
-  }
+// 34567890123456789012345678901234567890123456789012345678901234567890123456789
 
-  //Add header 2 and hyperlink  in Document to trello task  
-  body.appendParagraph(trelloCardName)
-      .setHeading(DocumentApp.ParagraphHeading.HEADING2).setLinkUrl(trellourl)
-  
-  //Get the timesheet link from the organisation spreadsheet
-  var orgData = SpreadsheetApp.openById(orgId).getSheetByName('Organisations').getDataRange()
+// JSHint - TODO
+/* jshint asi: true */
+
+(function() {"use strict"})()
+
+// Code review all files - TODO
+// JSHint review (see files) - TODO
+// Unit Tests - TODO
+// System Test (Dev) - TODO
+// System Test (Prod) - TODO
+
+// Config.gs
+// =========
+//
+// Dev: AndrewRoberts.net
+//
+// All the constants and configuration settings
+
+// Configuration
+// =============
+
+var SCRIPT_NAME = "TsJournalTrello"
+var SCRIPT_VERSION = "v0.1.dev_dlt"
+
+var PRODUCTION_VERSION_ = false
+
+// Log Library
+// -----------
+
+var DEBUG_LOG_LEVEL_ = PRODUCTION_VERSION_ ? Log.Level.INFO : Log.Level.FINER
+var DEBUG_LOG_DISPLAY_FUNCTION_NAMES_ = PRODUCTION_VERSION_ ? Log.DisplayFunctionNames.NO : Log.DisplayFunctionNames.YES
+
+// Assert library
+// --------------
+
+var SEND_ERROR_EMAIL_ = PRODUCTION_VERSION_ ? true : false
+var HANDLE_ERROR_ = Assert.HandleError.THROW
+var ADMIN_EMAIL_ADDRESS_ = ''
+
+// Tests
+// -----
+
+var TEST_SHEET_ID_ = ''
+
+// Constants/Enums
+// ===============
+
+
+
+// Function Template
+// -----------------
+
+/**
+ *
+ *
+ * @param {object} 
+ *
+ * @return {object}
+ */
+function tsJournalTrello(DocId) {
+
+  //Open the organisation spreadsheet
+  var orgData = SpreadsheetApp.openById(ORG_ID_)
+      .getSheetByName('Organisations')
+      .getDataRange()
   
   var data = orgData.getValues()
   var header = data.shift()
-  var timesheetUrl = ''
+  var timesheetUrl = null
+  var trelloBoardUrl = null
+
+  // Get the organisation name from the journal name, text before '- journal' 
+  var str = ' - Journal'
+  var CHAR_TO_END_ = str.length
+  var orgName = DocumentApp.getActiveDocument().getName().slice(0, -CHAR_TO_END_)
+    
+  //Loop through the rows of data
+  data.some(function(row)  {
+      
+    //check if the company name matches the orgName and if so, get the timesheet and
+    //Trello Board Url
+    var orgNameSearch = row[ORG_NAME_COL_]
+      
+    if (orgNameSearch === orgName) {
+      timesheetUrl = row[ORG_TS_COL_]
+      trelloBoardUrl = row[TRELLO_COL_]
+      return
+      } 
+    })    
+    
+    Logger.log('Timesheet URL: ' + timesheetUrl)
```

Comment:
Add-ons aren't supposed to have any Logger.log()s in. So to get into that habit remove all these before release (it is an item in the release checklist), and use BBLog as shown in GASFramework to log to a GSheet.